internal/controller/atlasipaccesslist/ipaccesslist.go

```diff
@@ -0,0 +1,152 @@
+package atlasipaccesslist
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/controller/workflow"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/ipaccesslist"
+)
+
+func (r *AtlasIPAccessListReconciler) handleIPAccessList(
+	ctx *workflow.Context,
+	ipAccessListService ipaccesslist.IPAccessListService,
+	projectID string,
+	ipAccessList *akov2.AtlasIPAccessList,
+) ctrl.Result {
+	akoIPAccessList, err := ipaccesslist.NewIPAccessListEntries(ipAccessList)
+	if err != nil {
+		return r.terminate(ctx, ipAccessList, api.IPAccessListReady, workflow.Internal, err)
+	}
+
+	atlasIPAccessList, err := ipAccessListService.List(ctx.Context, projectID)
+	if err != nil {
+		return r.terminate(ctx, ipAccessList, api.IPAccessListReady, workflow.Internal, err)
+	}
+
+	existInAtlas := len(atlasIPAccessList) > 0
+
+	if !ipAccessList.GetDeletionTimestamp().IsZero() {
+		if existInAtlas {
+			r.Log.Infof("deleting ip access list from project %s", projectID)
+			return r.deleteIPAccessList(ctx, ipAccessListService, ipAccessList, projectID, atlasIPAccessList, false)
+		}
+
+		r.Log.Info("releasing ip access list resource for deletion")
+		return r.unmanage(ctx, ipAccessList)
+	}
+
+	if !existInAtlas {
+		r.Log.Infof("creating ip access list on project %s", projectID)
+		return r.createIPAccessList(ctx, ipAccessListService, ipAccessList, projectID, akoIPAccessList)
+	}
+
+	r.Log.Infof("managing ip access list on project %s", projectID)
+	return r.manageIPAccessList(ctx, ipAccessListService, ipAccessList, projectID, akoIPAccessList, atlasIPAccessList)
```

Comment:
nit: this continues to be the discussion from above, we have a `handleIPAccessList`, and now a `manageIPAccessList`. it is not clear from the state machine what this is.

Reply:
suggestion: inline this into `handleIPAccessList`, that way we have the logic for the initial state in one place.